app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/AMLSController.scala

```diff
@@ -52,7 +53,25 @@ class AMLSController @Inject()(
   val showMoneyLaunderingComplianceForm: Action[AnyContent] = Action.async { implicit request =>
     withSubscribingAgent { _ =>
       withManuallyAssuredAgent {
-        Future.successful(Ok(html.money_laundering_compliance(amlsForm(amlsBodies.keys.toSet), amlsBodies)))
+        for {
+          cachedAmlsDetails <- sessionStoreService.fetchAMLSDetails
+          cachedGoBackUrl   <- sessionStoreService.fetchGoBackUrl
+        } yield {
+          (cachedAmlsDetails, cachedGoBackUrl) match {
+            case (Some(amlsDetails), mayBeGoBackUrl) =>
+              val form: Map[String, String] =
+                Map(
+                  "amlsCode"         -> amlsBodies.find(_._2 == amlsDetails.supervisoryBody).map(_._1).getOrElse(""),
+                  "membershipNumber" -> amlsDetails.membershipNumber,
+                  "expiry.day"       -> amlsDetails.membershipExpiresOn.getDayOfMonth.toString,
+                  "expiry.month"     -> amlsDetails.membershipExpiresOn.getMonthValue.toString,
+                  "expiry.year"      -> amlsDetails.membershipExpiresOn.getYear.toString
+                )
+              Ok(html.money_laundering_compliance(amlsForm(amlsBodies.keySet).bind(form), amlsBodies, mayBeGoBackUrl))
```

Comment:
this pre-populates the amls details already stored in the cache (if there are any)

Reply:
and also passing the goBack url (if stored in session already)